File: rb/features.rb
# Chocomilk SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module ChocomilkFeatures
  def self.make_feature(name)
    case name
    when "base"
      ChocomilkBaseFeature.new
    when "ratelimit"
      ChocomilkRatelimitFeature.new
    when "retry"
      ChocomilkRetryFeature.new
    when "test"
      ChocomilkTestFeature.new
    when "timeout"
      ChocomilkTimeoutFeature.new
    else
      ChocomilkBaseFeature.new
    end
  end
end

File: py/features.py
# Chocomilk SDK feature factory

from feature.base_feature import ChocomilkBaseFeature
from feature.test_feature import ChocomilkTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ChocomilkBaseFeature(),
        "test": lambda: ChocomilkTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
